# Patch to fix build and install, libwebp dynamically
# Copyright (C) 2010 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# Visit SlackPack at http://sotirov-bg.net/slackpack/
#
# $Id: webpconv-0.0.1.diff,v 1.1 2010/10/17 09:23:06 gsotirov Exp $
#
diff -urNad libwebp-orig/makefile libwebp/makefile
--- libwebp-orig/makefile	2010-09-24 07:31:28.000000000 +0300
+++ libwebp/makefile	2010-10-17 12:19:48.000000000 +0300
@@ -78,18 +78,22 @@
 
 ######################################################################
 
-all:	$(SRC:%.c=%)
+all: webpconv
 
 
 ######################################################################
 
-webpconv: libwebp.o webpconv.o $(LEPTLIB)
-	$(CC) -o webpconv webpconv.o $(ALL_LIBS) libwebp.o $(VPX_LIB)
+webpconv: webpconv.o $(LEPTLIB)
+	$(CC) -o webpconv webpconv.o $(ALL_LIBS) -lwebp
 
-libwebp.o: webpimg.c
-	$(CC) -c $(CPPFLAGS) -o libwebp.o webpimg.c
+webpconv.o: webpconv.c
+	$(CC) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
+
+install:
+	mkdir -p $(DESTDIR)$(prefix)/bin
+	install -m 755 webpconv $(DESTDIR)$(prefix)/bin/webpconv
 
 clean:
-	rm webpconv *.o
+	rm -r webpconv *.o
 
 ######################################################################
